pkg/activator/handler/revision_handler.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"knative.dev/pkg/network"
+	"knative.dev/serving/pkg/activator"
+)
+
+// RevisionHandler is a handler that decode the revision name and namespace from the Host header.
+type RevisionHandler struct {
+	NextHandler http.Handler
+	Domain      string
+}
+
+// NewRevisionHandler creates a handler that if the headers aren't explicitly specified,
+// then decode the revision name and namespace from the Host header and set it to the header.
+func NewRevisionHandler(h http.Handler) http.Handler {
+	return &RevisionHandler{
+		NextHandler: h,
+		Domain:      network.GetClusterDomainName(),
+	}
+}
+
+func (h *RevisionHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	namespace := r.Header.Get(activator.RevisionHeaderNamespace)
+	name := r.Header.Get(activator.RevisionHeaderName)
+	if name != "" && namespace != "" {
+		h.NextHandler.ServeHTTP(rw, r)
+		return
+	}
+
+	if n, ns, ok := getNameAndNamespace(r.Host, h.Domain); ok {
+		r.Header.Set(activator.RevisionHeaderNamespace, ns)
+		r.Header.Set(activator.RevisionHeaderName, n)
+		h.NextHandler.ServeHTTP(rw, r)
+		return
+	}
```

Comment:
I don't think we need to do this. In request_log.go we can get the revision using [`revisionFrom(req.getContext())`](https://github.com/knative/serving/blob/master/pkg/activator/handler/context.go#L42), and use the name and namespace from that revision to log the request. We can do that - I think - without any changes to context_handler.go (it [already stores the revision and the revision name/namespace in the context via `withRevision()` and `withRevID()`](https://github.com/knative/serving/blob/master/pkg/activator/handler/context_handler.go#L79-L80)).

Reply:
I see. updated.